### PR TITLE
Feat kafka env update

### PR DESCRIPTION
**Pull request by LazyBST**
[Notion task](https://www.notion.so/o4s/Migrate-all-Kafka-workloads-to-Strimzi-cluster-3437e42e60ba4add86f54413f549f2ee)

# Change
- Moved kakfa env and its checks from service level to writer-dev.

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -82,35 +47,35 @@ export class KafkaProducer {
   get producer() {
     return this._producer;
   }
+
+  publish = async (
+    producerInput: IProducerInput,
+    logger?
```
Comment (jonscyr) [inline src/app.ts:53]: i think you can give a default value to logger here.

logger = console. 

in which case, you dont have to have fallbacks in the code.

File: src/config/writer.config.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+]
+
+let config: Record<string, any> = {};
+
+export const setEnvironment = () => {
+  /** load environment variables */
+  const loggerRootDir = dirname(dirname(__dirname));
+  const applicationRootDir = dirname(dirname(dirname(loggerRootDir)));
+
+  const envFilePath = path.resolve(applicationRootDir, "environment", ".env");
```
Comment (jonscyr) [inline src/config/writer.config.ts:26]: you dont have to read from env file.. just read from process.env. loading .env should be app's job.. just make sure in the app, .env is called first thing on app startup.

File: src/app.ts (hunk 1 above)
Comment (LazyBST) [reply src/app.ts:53]: Passing console as default argument will break type for logger. Below is the error displayed.

`Argument of type 'typeof logger' is not assignable to parameter of type 'Console'.
  Type 'typeof logger' is missing the following properties from type 'Console': assert, clear, count, countReset, and 18 more.`

One way to achieve this is
logger: any = console

But involves any type

File: src/config/writer.config.ts (hunk 2 above)
Comment (LazyBST) [reply src/config/writer.config.ts]: Updated it

File: src/app.ts (hunk 1 above)
Comment (jonscyr) [reply src/app.ts:53]: hmmm, ok, leave it then.. another day..

**Pull request merged**